extensions/indexes/lucene/src/org/exist/indexing/lucene/AnalyzerConfig.java

```diff
@@ -325,122 +326,150 @@ private static KeyTypedValue getConstructorParameter(Element param) throws Param
         final NamedNodeMap attrs = param.getAttributes();
 
         // Get name of parameter, NULL when no value is present
-        final String name;
-        if (attrs.getNamedItem(NAME_ATTRIBUTE) != null) {
-            name = attrs.getNamedItem(NAME_ATTRIBUTE).getNodeValue();
-        } else {
-            // DW: TODO need to check if the NULL value is safe to use.
-            name = null;
-        }
-        
+        Node namedItem = attrs.getNamedItem(NAME_ATTRIBUTE);
```

Comment:
Done on purpose ; goal is to make at little changes as possible because @ljo will provide a lucene update soon. i think the Optional() might things a bit more difficult to merge in later....